cli/lsp/text.rs

```diff
@@ -20,7 +20,7 @@ where
   P: FnMut(&T) -> bool,
 {
   let mut left = 0;
-  let mut right = slice.len();
+  let mut right = slice.len() - 1;
```

Comment:
What did you specifically run into that needed this logic updated?  It would be to get an example of what wasn't working as a test that the logic update makes sense.

Reply:
Some test were failing: `lsp::text::tests::test_get_edits` and `lsp::text::tests::test_line_index`
For test_line_index this one in particular is was returning `line:2 character:0` without this change
```
assert_eq!(
      index.position_utf16(11.into()),
      lsp::Position {
        line: 1,
        character: 5
      }
    );
```
Now that `utf8_offsets` and `utf16_offsets` length is equal to `# of lines + 1` (because I added a last `push` call after the for loop so we can calculate the total text size) then `slice.len() - 1` is equal to `slice.len()` before my change